### PR TITLE
Add testing of packages in a simulated deployed state

### PR DESCRIPTION
**Pull request by Mearman**
This PR adds testing of packages in a deployed state without fully deploying them to npmjs. Currently it focuses on the `cli` and `create` packages.

The intention is to ensure functionality in clean/isolated environments such as that of a brand-new user with no breadboard packages installed locally.

The test fixtures test several entry points into the project, including several "mistakes" by new users that would be possible speed bumps and could be smoothed out with some aliased packages. That is discussed further in https://github.com/breadboard-ai/breadboard/issues/674.

This work started at https://github.com/breadboard-ai/breadboard/issues/633, where I discovered that some dependencies were not correctly resolved when using the `cli` in a fresh environment.

This PR reverts the changes I previously made to the dependency handling of the `cli` package.
It also fixes an issue I discovered while adding these tests whereby in `create-breadboard-kit`, `dist/assets` was not being created. However, as @PaulKinlan points out, it may be an issue with the order of operations. So, I would be happy to remove that from this PR.

PR https://github.com/breadboard-ai/breadboard/pull/677 tests the **already deployed** packages and demonstrates some of the scenarios and issues I've discovered more cleanly. I also improved the test fixture handling process, which I will bring across to this workflow.

For each of the packages to be tested in an artificially-deployed state, this pipeline performs the following actions:
- Update the name of the package to match the scope required by the GitHub NPM registry
    - This also will prevent incorrectly resolving the "real" packages.
- Set the versions to low pre-release versions to reduce the chance of confusion with production packages.
    - For a while, I did have an unpublish step, but then I realised having them accessible is also helpful for local testing.
- Update the dependencies of each package to point to aliases of the packages that have been renamed for testing. e.g.:
```
"@google-labs/breadboard": "npm:@breadboard-ai/breadboard:*"
```
- The packages are then built and deployed to the GitHub npm registry.
- The actual test CI job then runs, which intentionally has no dependencies already installed
    - This job has a matrix of script and version fixtures that test several scenarios
    - These fixtures `install` and or `exec` the packages that had been artificially deployed to the GitHub registry
    - As mentioned above, I will likely improve the test process based on the pipeline from PR [#677](https://github.com/breadboard-ai/breadboard/pull/677)

Most of the repackaging logic is implemented in TypeScript and bundled for execution rather than working within the constraints of YAML workflow syntax.

### CODE REVIEW

**Comment by Mearman** [comment]
nearly 200 commits later...

**Comment by Mearman** [comment]
@dglazkov I think this is good to merge now. I can squash the issues discovered by it in a new branch

**Comment by aomarks** [comment]
Could you explain the high level goals here? Sorry if I missed some Discord conversations. My first reaction is: this seems like a lot of code, and I'm not quite sure what problem it's solving.

**Comment by Mearman** [comment]
@aomarks sorry the information is somewhat dispersed as it's crossed over a few different issues and PRs. From several cases, I found out that the `create` and `cli` packages aren't working in isolation.

The pipeline now allows testing of various `npx exec` usages in completely clean environments.

A lot of the code is to do the chopping and changing is for the aliasing of the packages so the GitHub registry accepts them and they don't resolve to the "real" dependencies.

I've managed to track down a few of the issues such as https://github.com/breadboard-ai/breadboard/pull/665/
The remaining issues I think are related to bundling.

reference:
https://github.com/ExaDev-io/breadboard/actions/runs/7769652556/job/21188959255
https://github.com/breadboard-ai/breadboard/issues/633

**Comment by Mearman** [comment]
this was a related discovery of this pipeline https://github.com/breadboard-ai/breadboard/issues/674

File: packages/create-breadboard-kit/package.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
   },
   "wireit": {
     "copy-assets": {
-      "command": "cp -r assets dist",
+      "command": "mkdir -p dist/ && cp -r assets/ dist/assets/",
```
Comment (PaulKinlan) [inline packages/create-breadboard-kit/package.json:22]: The dist folder should already be created iirc - if it's not it suggests that there is an ordering of the build scripts issue... (also the trailing / is not needed.)

File: packages/breadboard-cli/vite.config.ts [hunk 2]
```diff
@@ -18,19 +18,6 @@ if (import.meta.resolve) {
 }
 
 export default defineConfig({
```
Comment (PaulKinlan) [inline packages/breadboard-cli/vite.config.ts:20]: Why is this being removed? We literally just added it in to fix an issue that was identified.

Comment (Mearman) [reply packages/breadboard-cli/vite.config.ts:20]: it didn't resolve the issue. but with this pipeline it'll allow testing for potential solutions.
see here for the issue persisting in the currently deployed version:
https://github.com/ExaDev-io/breadboard/actions/runs/7817291975/job/21324831812#step:5:27

**Comment by Mearman** [comment]
Yes. I should have at least left these in draft until I'd collected my thoughts and actually described things in full.
I shall go and actually add descriptions

**Comment by PaulKinlan** [comment]
Spent some time checking it out. Some more comments:

1) The huge list of files for the commands is kinda distracting, personally I would just have a main.ts and then commands.ts with all of the commands exported. 
2) I worry about the publish step (even if it is to github). I was wondering if instead of `publish` we `npm pack` the packages we need to test when breadboard is getting built and store the assets, then in the `test` step we don't run `actions/checkout` but instead just download the assets and install them `npm i <tar file>` them in the fresh node environment.

**Comment by Mearman** [comment]
Thank you for taking the time to look through.
I like the sound of that idea @PaulKinlan. It would be much more simple. I shall mark this PR as draft for the time being and have an experiment.

**Pull request closed without merging**